service/matching/handler/engine.go

```diff
@@ -598,18 +597,17 @@ pollLoop:
 			return nil, err
 		}
 
-		e.emitForwardedFromStats(hCtx.scope, task.IsForwarded(), req.GetForwardedFrom())
-
 		if task.IsStarted() {
 			// tasks received from remote are already started. So, simply forward the response
 			return task.PollForActivityResponse(), nil
 		}
+		e.emitForwardedFromStats(hCtx.scope, task.IsForwarded(), req.GetForwardedFrom())
+		e.emitTaskIsolationMetrics(hCtx.scope, task.Event.PartitionConfig, req.GetIsolationGroup())
```

Comment:
Just curious, what's the reason for moving the taskIsolationMetric emitting here?

Reply:
I just updated the description of the PR.

Reply:
Thank you